src/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -148,7 +172,7 @@ public IPAddress(byte[] address)
         // and winsock only understands Int32.
         internal IPAddress(int newAddress)
         {
-            Address = (long)newAddress & 0x00000000FFFFFFFF;
+            _addressOrScopeId = (uint)newAddress & 0x00000000FFFFFFFF;
```

Comment:
The mask is now just pointless. Its original purpose was to cause `(long)newAddress` to behave as though it were `(long)(uint)newAddress`. In the new code, `(uint)newAddress` is sufficient.